File: Raspi/Freenove_RFID_Starter_Kit_for_Raspberry_Pi/Code/Python_GPIOZero_Code/03.1.1_LightWater/LightWater.py
#!/usr/bin/env python3
########################################################################
# Filename    : LightWater.py
# Description : Use LEDBar Graph(10 LED) 
# Author      : www.freenove.com
# modification: 2023/05/11
########################################################################
from gpiozero import LED
from time import sleep

ledPins = [17, 18, 27, 22, 23, 24, 25, 2, 3, 8]
leds = [LED(pin=pin) for pin in ledPins] 
    
def loop():
    while True:
        for index in range(0,len(ledPins),1):      # make led(on) move from left to right
            leds[index].off()  
            sleep(0.1)
            leds[index].on() 
        for index in range(len(ledPins)-1,-1,-1):   #move led(on) from right to left
            leds[index].off()  
            sleep(0.1)
            leds[index].on() 

if __name__ == '__main__':     # Program entrance
    print ('Program is starting...')
    try:
        loop()
    except KeyboardInterrupt:  # Press ctrl-c to end the program.
        print("Ending program")
    finally:
        for index in range(0,len(ledPins),1): 
            leds[index].close()  

File: Raspi/Freenove_RFID_Starter_Kit_for_Raspberry_Pi/Code/C_Code/03.1.1_LightWater/LightWater.c
/**********************************************************************
* Filename    : LightWater.c
* Description : Use LEDBar Graph(10 LED) 
* Author      : www.freenove.com
* modification: 2019/12/27
**********************************************************************/
#include <wiringPi.h>
#include <stdio.h>

#define ledCounts 10
int pins[ledCounts] = {0,1,2,3,4,5,6,8,9,10};

void main(void)
{
	int i;
	printf("Program is starting ... \n");
	
	wiringPiSetup(); //Initialize wiringPi.
	
	for(i=0;i<ledCounts;i++){       //Set pinMode for all led pins to output
		pinMode(pins[i], OUTPUT);		
	}
	while(1){
		for(i=0;i<ledCounts;i++){   // move led(on) from left to right
			digitalWrite(pins[i],LOW);
			delay(100);
			digitalWrite(pins[i],HIGH);
		}
		for(i=ledCounts-1;i>-1;i--){   // move led(on) from right to left
			digitalWrite(pins[i],LOW);
			delay(100);
			digitalWrite(pins[i],HIGH);
		}
	}
}

